services/autorust/codegen/src/codegen.rs

```diff
@@ -503,4 +539,12 @@ mod tests {
         assert_eq!("bytes :: Bytes", tp.to_string());
         Ok(())
     }
+
+    #[test]
+    fn test_with_union() -> anyhow::Result<()> {
+        let mut tp = TypeNameCode::try_from("farm::Animal")?;
+        tp.union(true);
+        ensure!("farm :: AnimalUnion" == tp.to_string());
```

Comment:
```suggestion
    fn test_with_union() -> Result<()> {
        let mut tp = TypeNameCode::try_from("farm::Animal")?;
        tp.union(true);
        assert_eq!("farm :: AnimalUnion", tp.to_string());
```

Reply:
Do you anticipate refactoring the results to use anyhow?  Otherwise, I think this might be reasoanble instead.

Reply:
I've been using `anyhow::Result<()>` in tests along with `ensure!` which uses results instead of panicking. I like it. I would be interested in changing over autorust tests.

Reply:
I can move the change to a separate PR if it is a blocker.

Reply:
I backed out the changes. I will make a separate PR if I wish to.